concent_api/core/payments/bankster.py

```diff
@@ -249,7 +224,8 @@ def settle_overdue_acceptances(
     requestor_ethereum_address: str,
     provider_ethereum_address: str,
     acceptances: List[SubtaskResultsAccepted],
-) -> ClaimPaymentInfo:
+    requestor_public_key: bytes,
+) -> Union[DepositClaim, str, None]:
```

Comment:
I have quite mixed feelings about this. I mean, we are returning 3 quite different types of objects here. My biggest concern is returning `DepositClaim` (normal operation) or `str` to indicate a `timestamp_error`. The last part seems to be a bussiness logic that should be handle in `message_handlers`. Maybe it would be better to raise some exception and handle it in the caller?

Reply:
Yeah. I've already suggested an enum (https://github.com/golemfactory/concent/pull/929#discussion_r232216556). which should be an improvement over returning a bare string but but you're right, the `"timestamp_error"` does feel like something that should be an exception.

Reply:
:+1: done